### DISCUSSION TITLE
[New Feature] Passable buildings

### DISCUSSION THREAD

**Discussion by Crimsonum** [Vinifera-Developers/Vinifera#1336] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

Building types that units can freely pass through.

For reference on how it could work, see `FirestormWall`, which allows any unit to pass through a deactivated Firestorm wall, or `Bib`, which renders the building's east row of cells passable to vehicles.

### Proposed Code (optional)

`Passable` (bool); if set, units can move through the building.

**Comment by Onidotmoe** [comment]
```
[General]
PassableTypes=Infantry,Vehicles,Aircraft
PassableType0=E1,other infantry units...
; Or group them by locomotive
PassableType0=SomeLocomotive

[SomeTree]
; Vehicles can't go onto tiles that this tree occupies.
Passable=Infantry,Aircraft
```
I can see this being especially useful for like allowing infantry to hide behin trees to avoid getting splattered by vehicles.

**Comment by Crimsonum** [comment]
Infantry can already move to cells occupied by trees, provided that the tree's OccupationBits allow this & that there are free cell spots. However, you can't directly order infantry to move to the cell occupied by a tree, instead you have to move infantry through it then order them to stop at the cell, which is very clumsy and should be rectified.
